__tests__/api.test.js

```diff
@@ -150,4 +151,60 @@ describe('GET /api/articles/:article_id/comments', () => {
                 expect(body.msg).toBe('Bad request');
             });
     });
+});
+
+describe('POST /api/articles/:article_id/comments', () => {
+    test("201: responds with the posted comment and adds the comment to an article", () => {
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'this article sucks'
+        };
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({body}) => {
+            expect(body.comment).toMatchObject({
+                author: 'butter_bridge',
+                body: 'this article sucks'
+            })
+        })
+    })
+    test("400: responds with an appropriate status and error message when provided with a non-existent user-name", () => {
+        const newComment = {
+            body: 'this article sucks'
+        };
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Username does not exist')
+        })
+    });
+    test("400: responds with an appropriate status and error message when provided with a non-existent body", () => {
+        const newComment = {
+            username: 'butter_bridge'
+        };
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Body does not exist')
+        })
+    })
+    test('404: responds with appropriate status and msg when given a valid id but non-existent id', () => {
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'this article sucks'
+        };
+        return request(app)
+        .post("/api/articles/999/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe('article does not exist')
+        });
```

Comment:
We could do with one more test here, for if we go to` /api/articles/banana/comments`